client/landing/stepper/declarative-flow/internals/steps-repository/automated-copy-site/index.tsx

```diff
@@ -0,0 +1,170 @@
+		setIsCopying,
+		setIsSuccess,
+		setIsError,
+	};
+}
+
+const AutomatedCopySite: Step = function AutomatedCopySite( { navigation } ) {
+	const { submit } = navigation;
+	const { __ } = useI18n();
+	const {
+		isCopying,
+		isLoading,
+		setIsCopying,
+		setIsSuccess,
+		setIsError,
+		isSuccess,
+		isError,
+		error,
+	} = useCopyingStatus();
+	const site = useSite();
+	const urlQueryParams = useQuery();
+	const siteSlug = urlQueryParams.get( 'siteSlug' );
+	const sourceSlug = urlQueryParams.get( 'sourceSlug' );
+	const sourceSite = useSelect(
+		( select ) => sourceSlug && select( SITE_STORE ).getSite( sourceSlug )
+	);
+	const sourceSiteId = sourceSite?.ID;
+	const { requestLatestAtomicTransfer } = useDispatch( SITE_STORE );
+	const { getSiteLatestAtomicTransfer, getSiteLatestAtomicTransferError } = useSelect( ( select ) =>
+		select( SITE_STORE )
+	);
+
+	useEffect( () => {
+		if ( isCopying && site ) {
+			const timer = setInterval( async () => {
+				await requestLatestAtomicTransfer( site.ID );
+				const transfer = getSiteLatestAtomicTransfer( site.ID );
+				const transferError = getSiteLatestAtomicTransferError( site.ID );
+				const transferStatus = transfer?.status;
+				const isTransferringStatusFailed = transferError && transferError?.status >= 500;
+
+				if ( isTransferringStatusFailed ) {
+					setIsError( 'Error Copying' );
+				} else if ( transferStatus === 'completed' ) {
+					setIsSuccess();
+				}
+			}, TIME_CHECK_TRANSFER_STATUS );
+			return () => clearTimeout( timer );
+		}
+	}, [
+		getSiteLatestAtomicTransfer,
+		getSiteLatestAtomicTransferError,
+		isCopying,
+		requestLatestAtomicTransfer,
+		setIsError,
+		setIsSuccess,
+		site,
+	] );
+
+	const startCopying = useCallback(
+		function startCopying() {
+			if ( ! site?.ID || ! sourceSiteId ) {
+				return;
+			}
+			return wpcom.req
+				.post( {
+					path: `/sites/${ site.ID }/copy-from-site`,
+					apiNamespace: 'wpcom/v2',
+					body: {
+						source_blog_id: sourceSiteId,
+					},
+				} )
+				.then( () => {
+					setIsCopying();
+				} )
+				.catch( ( e: { message: string; code: string; data: { status: number } } ) => {
+					// eslint-disable-next-line no-console
+					console.error( e );
+					setIsError( e.message || 'Error Starting the Copy' );
+				} );
+		},
+		[ setIsCopying, setIsError, site?.ID, sourceSiteId ]
+	);
+
+	useEffect( () => {
+		startCopying();
+	}, [ startCopying ] );
+
+	if ( ! sourceSlug || ! siteSlug ) {
+		// TODO: show a better error or redirect to the start /sites
+		return <h1>{ __( 'Missing required parameters' ) }</h1>;
+	}
+
+	if ( ! sourceSiteId || ! site ) {
+		// eslint-disable-next-line no-console
+		console.info( { sourceSlug, siteSlug, sourceSiteId, site } );
+		// Wait for stores to populate
+		return null;
+	}
+
+	return (
+		<StepContainer
+			stepName="AutomatedcopySite"
+			hideBack
+			recordTracksEvent={ recordTracksEvent }
+			stepContent={
+				<div>
+					{ isLoading && <h1>{ __( 'Copying Site…' ) }</h1> }
```

Comment:
Created an issue for improving this: https://github.com/Automattic/dotcom-forge/issues/1480

Reply:
I've added an updated screenshot to https://github.com/Automattic/dotcom-forge/issues/1480